pkg/apps/util/scheme.go

```diff
@@ -0,0 +1,46 @@
+package util
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	coreapi "k8s.io/kubernetes/pkg/apis/core"
+
+	appsv1 "github.com/openshift/api/apps/v1"
+	appsapi "github.com/openshift/origin/pkg/apps/apis/apps"
+	appsv1helpers "github.com/openshift/origin/pkg/apps/apis/apps/v1"
+)
+
+var (
+	// for decoding, we want to be tolerant of groupified and non-groupified
+	annotationDecodingScheme = runtime.NewScheme()
+	annotationDecoder        runtime.Decoder
+
+	// for encoding, we want to be strict on groupified
+	annotationEncodingScheme = runtime.NewScheme()
+	annotationEncoder        runtime.Encoder
+)
+
+func init() {
+	utilruntime.Must(corev1.AddToScheme(annotationDecodingScheme))
+	utilruntime.Must(appsv1.AddToScheme(annotationDecodingScheme))
+	utilruntime.Must(appsv1helpers.AddToScheme(annotationDecodingScheme))
```

Comment:
move internal stuff into the TODO block to remove when internal versions go away